app/models/rent.rb

```diff
@@ -2,4 +2,12 @@ class Rent < ApplicationRecord
   belongs_to :user
   belongs_to :book
   validates :user, :book, :initial_date, :final_date, presence: true
+
+  def valid_dates?
+    book_rents = Rent.where('book_id = ?', book_id)
+    book_rents.delete(self)
+    book_rents.all? do |previous_rent|
```

Comment:
Very low efficiency here.

Reply:
This was part of the extra PR. I put it here by accident so I'll delete it for this PR.